TheMET/TheMET/Network/NetworkManager.swift

```diff
@@ -10,40 +10,61 @@ import Foundation
 class NetworkManager {
     
     func get(urlString: String, parameters: [String : String], completion: @escaping ([String: Any]?) -> Void)  {
-        guard var components = URLComponents(string: urlString) else {
-            completion(nil)
-            return
-        }
-        components.queryItems = parameters.map { (key, value) in
-            URLQueryItem(name: key, value: value)
-        }
-        components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
-        guard let url = components.url else {
-            completion(nil)
-            return
-        }
-        var urlRequest = URLRequest(url: url)
-        urlRequest.httpMethod = "GET"
-        let urlSession = URLSession.shared
-        let fileDownloadTask = urlSession.dataTask(
-            with: urlRequest
-        ) { data, _, error in
-            let executeCompletionOnMain: ([String: Any]?) -> Void = { loadedData in
-                DispatchQueue.main.async {
-                    completion(loadedData)
-                }
+        var lastRequestTime = Date()
+            guard var components = URLComponents(string: urlString) else {
+                completion(nil)
+                return
+            }
+            components.queryItems = parameters.map { (key, value) in
+                URLQueryItem(name: key, value: value)
             }
-            guard error == nil,
-                  let data = data else {
-                executeCompletionOnMain(nil)
+            components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
+            guard let url = components.url else {
+                completion(nil)
                 return
             }
-            let result = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
-            executeCompletionOnMain(result)
+            var urlRequest = URLRequest(url: url)
+            urlRequest.httpMethod = "GET"
+            let urlSession = URLSession.shared
+            let fileDownloadTask = urlSession.dataTask(
+                with: urlRequest
+            ) { data, _, error in
+                let executeCompletionOnMain: ([String: Any]?) -> Void = { loadedData in
+                    DispatchQueue.main.async {
+                        completion(loadedData)
+                    }
+                }
+                guard error == nil,
+                      let data = data else {
+                    executeCompletionOnMain(nil)
+                    return
+                }
+                let result = try? JSONSerialization.jsonObject(with: data) as? [String: Any]
+                executeCompletionOnMain(result)
+            }
+        while !self.timeIntervalBWLastRequestTimeToNow(lastRequest: lastRequestTime) {
+            DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + .milliseconds(Int(12.5)), qos: .background) {
```

Comment:
This code does not postpone the network request. The `DispatchQueue.global().asyncAfter` does nothing because you didn't pass what exact command should be executed after 12.5 sec.

Instead this method just goes to a `while` loop, and executes this dummy `DispatchQueue.global().asyncAfter` call and blocks the current code execution for 12.5ms. It means that the whole App is freezing for 12.5ms. This is wrong, you shouldn't block code execution by going to infinitive loop.